app/assets/javascripts/listings/ListingService.js.coffee

```diff
@@ -606,7 +606,8 @@ ListingService = ($http, $localStorage, $q, $state, $translate, ModalService, Ex
     Service.stubListingPreferences()
     # if this listing had stubbed preferences then we can abort
     if !_.isEmpty(Service.listing.preferences)
-      return $q.when(Service.listing.preferences)
+      return $q.when(Service.listing.preferences).then ->
```

Comment:
this function is expecting a promise to be returned, is this still returning a promise if I tack on a then? Since it's not really waiting on anything I could probably just set the loading to false in the line above. thoughts?